native/more/friend-list.react.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+
+import React from 'react';
+import { View, Text, TouchableOpacity, FlatList } from 'react-native';
+
+import { connect } from 'lib/utils/redux-utils';
+import { type UserInfo } from 'lib/types/user-types';
+
+import type { NavigationRoute } from '../navigation/route-names';
+import type { AppState } from '../redux/redux-setup';
+import { AddFriendsModalRouteName } from '../navigation/route-names';
+import { styleSelector } from '../themes/colors';
+
+import type { MoreNavigationProp } from './more.react';
+import FriendListItem from './friend-list-item.react';
+
+type Props = {|
+  navigation: MoreNavigationProp<'FriendList'>,
+  route: NavigationRoute<'FriendList'>,
+  // Redux state
+  styles: typeof styles,
+|};
+
+type State = {||};
+
+class FriendList extends React.PureComponent<Props, State> {
+  render() {
+    return (
+      <View style={this.props.styles.container}>
+        <TouchableOpacity
+          onPress={() => {
+            this.props.navigation.navigate({
+              name: AddFriendsModalRouteName,
+            });
+          }}
+        >
+          <Text>add</Text>
+        </TouchableOpacity>
+        <FlatList
+          data={[
+            { id: '1', username: 'example' },
+            { id: '2', username: 'example2' },
+          ]}
+          renderItem={this.renderItem}
+        />
+      </View>
+    );
+  }
+
+  renderItem = ({ item }: { item: UserInfo }) => {
+    return <FriendListItem userInfo={item} />;
+  };
+}
+
+const styles = {
+  container: {
+    flex: 1,
+  },
+};
+const stylesSelector = styleSelector(styles);
+
+export default connect((state: AppState) => ({
+  styles: stylesSelector(state),
```

Comment:
The point of `stylesSelector` is actually only about color theming. It basically just replaces any of the strings in `native/themes/colors.js` with the corresponding color depending on whether the user is in dark mode or light mode. In this case you're not using any colors, so I would avoid using Redux and just reference the `styles` object directly (unless you have plans to add colors to `styles`)

Reply:
Have you ever considered using one of `css-in-js` libraries like [styled-components](https://styled-components.com/)? It works well on web and react-native, supports theming out of the box, and allows to use css syntax in react-native. JSX is much clearer since you don't have to pass `style` props every time and to have conditions:

```javascript
// App.js

const theme = {
  dark: {
     boxBackground: '#ffffff',
  },
  light: {
     boxBackground: '#000000',
  }
}

function App() {
  return (
     <ThemeProvider theme={dark ? theme.dark : theme.light}>
       <Screen />
     <ThemeProvider>
  )
}
```

```javascript
// Screen.js

import styled, { css } from 'styled-components/native';

...
function Screen(props) {
  return (
     <Box primary onLayout={() => console.log('I accept View props')}>
        <Title>I'm a Text but I have a title styles</Title>
     </Box>
  )
}

const Box = styled.View`
  background-color: ${props => props.theme.boxBackground}; // 1
  border: 1px solid ${props => props.theme.boxBorder}; // 2
  padding: 10px 5px; // 3

  ${props => props.primary && css`
    background-color: ${props => props.theme.boxPrimaryBackground};
    border-color: ${props => props.theme.boxPrimaryBorder};
  `} // 4
`;

// 1. It's easy to interpolate theme provided by ThemeProvider. It can be dynamic
// 2. It allows using web syntax in native for rules like border or box-shadow
// 3. It allows using px units in native which helps with writing responsive views for different device screen sizes
// 4. It allows to pass custom props and to avoid handling conditions in JSX
```

It supports SCSS syntax on the web (it allows to easily override styles of third-party components etc):
``` javascript
import styled, { css } from 'styled-components';

const Container = styled.div`
  background-color: 'blue';

  .inner {
     background-color:  'red';

     p {
        color: 'green';
     }
  }
`

<Container>
   <div class="inner">
      <p>Hello world</p>
   </div>
</Container>
```

Reply:
I've considered it before but never made the jump. I think this is a good idea, created #55 to track. If it's easy to set up feel free to do it sooner than later, we can handle refactoring the codebase piece-by-piece